src/wp-admin/includes/class-wp-plugin-dependencies.php

```diff
@@ -0,0 +1,664 @@
+<?php
+/**
+ * Dependencies manager for plugins.
+ *
+ * @package dependencies-manager.
+ * @since 5.9.0
+ */
+
+/**
+ * Plugins dependencies manager.
+ *
+ * @since 5.9.0
+ */
+class WP_Plugin_Dependencies {
+
+	/**
+	 * The database option where we store the array of plugins that should be active
+	 * but are not due to unmet dependencies.
+	 *
+	 * @since 5.9.0
+	 * @access private
+	 *
+	 * @var string
+	 */
+	const PENDING_PLUGIN_ACTIVATIONS_OPTION = 'pending_plugin_activations';
+
+	/**
+	 * Installed plugins.
+	 *
+	 * @since 5.9.0
+	 * @access private
+	 *
+	 * @var array[]
+	 */
+	private $installed_plugins;
+
+	/**
+	 * An array of admin notices to show.
+	 *
+	 * @since 5.9.0
+	 * @access private
+	 *
+	 * @var array<int, array>
```

Comment:
It looks like we don't use this kind of typed notation in the rest of core yet.

This might need some investigation to see if the DevHub parser supports it, and whether we could also start using it in the rest of core.

Reply:
This is taken from https://phpstan.org/writing-php-code/phpdoc-types#general-arrays. I've seen `array[]`, and `string[]` being used, so I assumed we can use the specific types in the arrays.

Reply:
@SergeyBiryukov I've set up a local wp dev env, with the parser for the DevHub on it. Do you have any tips on how I may test if this will work? I'd love to have more type specifics in the core, it will future proof it, and if we want to add phpstan checks in the future, it will help out a lot.